leftwm-core/src/handlers/command_handler/scratchpad_handler.rs

```diff
@@ -0,0 +1,1211 @@
+        .workspace(&manager.state.workspaces)
+    {
+        manager
+            .state
+            .windows
+            .iter()
+            .find(|w| ws.is_managed(w))
+            .map(|w| w.handle)
+    } else {
+        None
+    };
+    if let Some(handle) = handle {
+        manager.state.handle_window_focus(&handle);
+    }
+
+    Ok(())
+}
+
+/// Makes a scratchpad window visible:
+/// Expects that the window handle is a valid handle to an invisible scratchpad window
+fn show_scratchpad<C: Config, SERVER: DisplayServer>(
+    manager: &mut Manager<C, SERVER>,
+    scratchpad_window: &WindowHandle,
+) -> Result<(), &'static str> {
+    log::trace!("Show scratchpad window {:?}", scratchpad_window);
+    let current_tag = &manager
+        .state
+        .focus_manager
+        .tag(0)
+        .ok_or("Could not retrieve the current tag")?;
+    let window = manager
+        .state
+        .windows
+        .iter_mut()
+        .find(|w| w.handle == *scratchpad_window)
+        .ok_or("Could not find window from scratchpad_window")?;
+    let previous_tag = window.tag;
+    window.untag();
+
+    // Remove the entry for the previous tag to prevent the scratchpad being
+    // refocused.
+    if let Some(previous_tag) = previous_tag {
+        manager
+            .state
+            .focus_manager
+            .tags_last_window
+            .remove(&previous_tag);
+    }
+    // Show the scratchpad.
+    window.tag(current_tag);
+
+    // Send tag changement to X
+    let act = DisplayAction::SetWindowTag(*scratchpad_window, window.tag);
+    manager.state.actions.push_back(act);
+    manager.state.sort_windows();
+    manager.state.handle_window_focus(scratchpad_window);
+    manager.state.move_to_top(scratchpad_window);
+
+    Ok(())
+}
+
+/// With the introduction of `VecDeque` for scratchpads, it is possible that a window gets destroyed
+/// in the middle of the `VecDeque`. This is an abstraction to retrieve the next valid pid from a
+/// scratchpad. While walking the scratchpad windows, invalid pids will get removed.
+fn next_valid_scratchpad_pid(
+    scratchpad_windows: &mut VecDeque<u32>,
+    managed_windows: &[Window],
+    direction: Direction,
+) -> Option<u32> {
+    while let Some(window) = if direction == Direction::Forward {
+        scratchpad_windows.pop_front()
+    } else {
+        scratchpad_windows.pop_back()
+    } {
+        if managed_windows.iter().any(|w| w.pid == Some(window)) {
+            if direction == Direction::Forward {
+                scratchpad_windows.push_front(window);
+            } else {
+                scratchpad_windows.push_back(window);
+            }
+            return Some(window);
+        }
+
+        log::info!(
+            "Dead window in scratchpad found, discard: window PID: {}",
+            window
+        );
+    }
+
+    None
+}
+
+/// Check if the scratchpad is visible on the current tag.
+/// If the name of the scratchpad is incorrect, false will be returned
```

Comment:
> /// If the name of the scratchpad is incorrect (...)

What do you mean with "is incorrect"? Do you just mean "if the scratchpad is not visible"?

Reply:
Incorrect as in not defined in the config. If the name is not in the config, it has no way of being initialized in the manager.

Reply:
Oh ok, that makes sense. I'd replace
> /// If the name of the scratchpad is incorrect, false will be returned

than with
```
/// Returns `false` if the scratchpad name isn't defined in the config
```